numalogic/registry/artifact.py

```diff
@@ -128,12 +128,8 @@ def construct_key(skeys: KEYS, dkeys: KEYS) -> str:
         return "::".join([_static_key, _dynamic_key])
 
 
-def _apply_jitter(ts: int, jitter_secs: int, jitter_steps_min: int):
-    return random.randrange(
-        min(ts - jitter_secs, ts),
-        ts + jitter_secs + 1,
-        60 * jitter_steps_min,
-    )
+def _apply_jitter(ts: int, jitter_sec: int, jitter_steps_min: int):
+    return random.randrange(max(ts - jitter_sec, ts), ts + jitter_sec + 1, 60 * jitter_steps_min)
```

Comment:
Thinking more on this, I think a generic alternative could be the following. That being said, I will leave the final call to you.

```
_apply_jitter(min_val: int, max_val: int, bins: int = 100):
  delta = max_val - min_val
  assert(0 < bins and bins < delta)
  return random.randrange(min_val, max_val, delta / bins)
```

Reply:
That change was still in works as I wanted to make it more robust :) .  Yes you are 100% correct on that.

These are the final changes I was looking for:
```
if jitter_sec == jitter_steps_min == 0:
        return ts
    if jitter_sec < 60 * jitter_steps_min:
        raise ConfigError("jitter_sec should be at least 60*jitter_steps_min")
    begin = ts if ts - jitter_sec < 0 else ts - jitter_sec
    end = ts + jitter_sec + 1
    return random.randrange(begin, end, 60 * jitter_steps_min)
```
a) The specific reason I do not want to keep it zero is because it defeats the purpose of TTL. The function should return a **Positive** value always. If random function returns 0 that means the TTL for storing becomes zero. That is misleading.  The above code will take care of that.

b) Let's say we pass a config such that ts = 1000, jitter_sec= 5 sec, and jitter_steps_min=2
In this scenario we will always return ts =1.  (always). This is the scenario you are referring to.  We have two options here. Either we throw a Configuration Error or we ignore it and jitter . Even though function will return the same value, it defeats the purpose of Jitter Conf. I have added a config check for that.
Here, we need to check : `(jitter_sec * 2) // (60 * jitter_steps_min) > 1` and throw config error.

c) There is another scenario such that ts = 1, jitter_sec= 5 sec, and jitter_steps_min=2.
In this case, the condition that should hold true is: "jitter_sec //(60 * jitter_steps_min)> 1"

From (2) and (3), if we union the conditions (3) holds true and hence you see the above condition. 

What do you think?

Reply:
The logic looks good.

Don't mean to prolong the completion of this PR but when you need to multiple a hardcode 60x to the jitter steps at every use of the variable. It may be simpler to keep the jitter range and step in the same (physical) dimension and unit. Again, it's something optional to consider.

Reply:
That is a fair ask.

No worries! I think I will have to wait till Monday for other's review. anyways :)